src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -640,7 +682,12 @@ bool dcd_edpt_open (uint8_t rhport, tusb_desc_endpoint_t const * desc_edpt)
   uint8_t const epnum = tu_edpt_number(desc_edpt->bEndpointAddress);
   uint8_t const dir   = tu_edpt_dir(desc_edpt->bEndpointAddress);
 
-  TU_ASSERT(epnum < ep_count);
+  #if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3, OPT_MCU_ESP32P4)
+  if (!dcd_edpt_available(rhport, dir)) {
+    TU_LOG(1, "No endpoints available (ep_max=%d) \r\n", dwc_ep_config[rhport].ep_max_count);
```

Comment:
Would have been good to make this an error, spent some time to figure out I ran out of endpoints as this 'silently' passes... Before there was an assert.

Reply:
Hi @bollenn, 
Sorry for the inconveniences. 

Actually, the whole idea of that MR was exactly as you have said: to display the message when there are no more EPs available.
Before, there was the assert, but the comparison itself was between wrong values, so the assert never worked, even in case when there no more vacant EPs available. 

May I ask, why is was "silent" in your case? TU_LOG was configured to 0?